stumpy/core.py

```diff
@@ -1971,3 +1971,81 @@ def _idx_to_mp(I, T, m, normalize=True):
     P[I < 0] = np.inf
 
     return P
+
+
+@njit(
+    # "(i8, i8, i8, i8[:], i8)"
+)
+def _get_tiles(m, n_A, n_B, diags, tile_size):
+    """
+    Split distance matrix into tiles of given size
+
+    Parameters
+    ----------
+    m : int
+        Window size
+
+    n_A : ndarray
+        The length of time series `T_A`
+
+    n_B : ndarray
+        The length of time series `T_B`
+
+    diags : ndarray
+        The diagonal indices of interest
+
+    tile_size : int
+        Maximum length of each tile
+
+    Returns
+    -------
+    tiles : ndarray
+        An array of 7-element-arrays representing each tile:
+        The first two values represent offset positions of the tile
+        The second two values represent dimensions of the tile
+        The third two values represent diagonal indices to traverse in the tile
+        The seventh value represents number of distances in the tile
+    """
+    # dimensions of distance matrix
+    dim = (n_A - m + 1, n_B - m + 1)
+    # number of tiles
+    tiles_count = np.uint64(np.ceil(dim[0] / tile_size) * np.ceil(dim[1] / tile_size))
+    tiles = np.zeros((tiles_count, 7), dtype=np.int64)
+
+    tile_idx = 0
+    # iterate over each tile
+    for y_offset in range(0, dim[0], tile_size):
+        # height of current tile
+        tile_height = min(tile_size, dim[0] - y_offset)
+        for x_offset in range(0, dim[1], tile_size):
+            # width of current tile
+            tile_width = min(tile_size, dim[1] - x_offset)
+
+            # lower and upper diagonal indices to traverse within tile
+            tile_lower_diag = max(1 - tile_height, diags[0] + y_offset - x_offset)
```

Comment:
Same reason I explained above, because of our exclusion zone, `diags` may start and end anywhere.

Reply:
Can you provide a concrete example of which combination of inputs (`m, n_A, n_B, diags, tile_size`) that would cause _get_tiles(m, n_A, n_B, diags, tile_size) to generate `tile_lower_diag = diags[0] + y_offset - x_offset` and when `tile_upper_diag = diags[-1] + 1 + y_offse - x_offset`?

I want to understand this better.